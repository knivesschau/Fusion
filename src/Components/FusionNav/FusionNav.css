/* mobile styles first, min 320px */
.Guest_Nav, .User_Nav {
    background-color: rgb(245, 245, 245, .9);
    box-shadow: inset 3px 0px 11px 0px black;
    padding: .25em;
    color: black;
    display: flex; 
    flex-direction: row;
    justify-content: space-around;
    text-align: center;
    font-size: 1.10rem;
    font-weight: bolder;
}

#link-home, #link-cookbook, #link-fusion, #link-logout, #link-register, #link-login {
    text-transform: uppercase;
    transition-property: background;
    transition-duration: .2s;
    transition-timing-function: ease-in;
}

#link-home:hover, #link-cookbook:hover, #link-fusion:hover, #link-logout:hover, #link-register:hover, #link-login:hover {
    cursor: pointer;
    text-decoration: underline;
    color: white;
    background-color: #acc29e;
}

#link-home:active, #link-cookbook:active, #link-fusion:active, #link-logout:active, #link-register:active, #link-login:active {
    background-color: #577147;
}

/* tablet and desktop styles */
@media only screen and (min-width: 768px) {
    .Guest_Nav, .User_Nav {
        font-size: 1.65rem;
    }
}

@media only screen and (min-width: 1200px) {
    .Guest_Nav, .User_Nav {
        font-size: 2rem;
    }
}

@media only screen and (min-width: 1600px) {
    .Guest_Nav, .User_Nav {
        font-size: 2.5rem;
        padding: .15em;
    }
} 
