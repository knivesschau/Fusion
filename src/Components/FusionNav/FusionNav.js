import React, { Component } from 'react';
import {Link} from 'react-router-dom';
import './FusionNav.css';

export default class FusionNav extends Component {
    render() {
        return (
            <div className="Fusion_Nav">
                <Link to="/">
                    Home
                </Link>

                <Link to="/register">
                    Sign Up
                </Link>

                <Link to="/login">
                    Login
                </Link>

                <Link to="/your-cookbook">
                    Cookbook
                </Link>

                <Link to="/starter-recipes">
                    Fuse
                </Link>
            </div>
        );
    }
}
