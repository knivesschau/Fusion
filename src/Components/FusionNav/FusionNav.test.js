import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import FusionNav from './FusionNav';

describe ('FusionNav Component', () => {
    it ('Renders the complete FusionNav Component and all nested Components', () => {
        const wrapper = shallow(<FusionNav/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});