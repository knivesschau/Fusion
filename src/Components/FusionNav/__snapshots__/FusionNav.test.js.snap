// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FusionNav Component Renders the complete FusionNav Component and all nested Components 1`] = `
<Fragment>
  <div
    className="Guest_Nav"
  >
    <Link
      id="link-home"
      to="/"
    >
      About
    </Link>
    <Link
      id="link-register"
      to="/get-started"
    >
      Register
    </Link>
    <Link
      id="link-login"
      to="/login"
    >
      Log In
    </Link>
  </div>
</Fragment>
`;
